### PR TITLE
Frontend: Added Login Page with modifications to other components

### PR DESCRIPTION
**Pull request by PrathamX595**
Added Login Page & Signup Page with other modifications

Fixes: #79

## Description
Added a login page with changes to some components namely:
1)Button component was changed to include `type` as a prop.
2)Input button was changed to adjust the roundness and the height and width props were made editable .
3)NavBar was modified to conditionally render the signin/login button.
4)Routes for Login and Signup pages were added.
5)Implemented authContext to the login page

## How to Test

- run the command `npm run dev`
- access the /login route

## Related Issues
#80 
#87 
#76 
## Checklist
- [x] I have tested these changes locally.
- [x] I have reviewed the code and ensured it follows the project's coding guidelines.
- [ ] I have updated the documentation, if necessary.
- [ ] I have assigned reviewers to this pull request.

## Screenshots (if applicable)
image

image

### CODE REVIEW

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -22,6 +23,8 @@ const router = createBrowserRouter([
             },
             { path: '/about', element: <About /> },
             { path: '/error', element: <Error /> },
+            { path: '/login', element: <Login />},
```
[the remark this answers is not in the document]
Comment (kuv2707) [reply frontend/src/App.tsx:26]: Let's go with login

[the remark this answers is not in the document]
Comment (PrathamX595) [reply frontend/src/App.tsx:26]: So the new users with no account will be directed to the login page ?

[the remark this answers is not in the document]
Comment (kuv2707) [reply frontend/src/App.tsx:27]: Oh i see
We'd use `sign-up` or `register` for that, not `sign-in`
signin and login mean the same thing

**Comment by kuv2707** [comment]
We won't be using google auth right now (that can be done later)
So change the text accordingly.

A technical note: Both login and signup are quite identical, but differ in the api call and that signup ( = for first time users) will have a confirm password field.

**Comment by kuv2707** [comment]
@PrathamX595 Login is implemented in the backend, so you can test if everything works as expected.
This PR should be done in priority, as only after this would we be able to proceed to the game page.

**Comment by PrathamX595** [comment]
> @PrathamX595 Login is implemented in the backend, so you can test if everything works as expected.
> This PR should be done in priority, as only after this would we be able to proceed to the game page.

I'll get it done latest by Tomorrow only a few changes are left

**Comment by PrathamX595** [comment]
@kuv2707 implemented the authContext let me know if anything else is needed to done

**Pull request merged**